### ISSUE TITLE
Suggestion: option to disable all special keys

### ISSUE THREAD

**Issue by emendelson** [joncampbell123/dosbox-x#568]
One of my DOSBox-based projects runs WordPerfect for DOS and related programs which use all the Ctrl-F# and Alt-F# combinations. I know I can use a mapper file to disable DOSBox-specific key combinations like Ctrl-F11, and I know I can disable these keys by commenting out lines in code. But for situations where I can fine-tune the settings in the conf file and have no real need for the special DOSBox keys, I wanted an option to turn off all those keys. 

So here's a crude brute-force patch that adds to [sdl] an enabledosboxkeys=false option that turns off those keys. There may be a thousand reasons not to use this, but I hope you might at least consider it:

https://github.com/emendelson/dosbox-x/commit/3c49924862e3ae2740566187a21a915f4e34513a

**Comment by sikthehedgehog** [comment]
I wonder if there should be a setting that disables all special shortcuts
*except* going windowed. Otherwise you risk being permanently locked out.

Some emulators for other systems that have keyboards provide a "capture
keyboard" combination instead. When captured all emulator shortcuts (except
that one for obvious reasons) do nothing and so can be passed onto the
emulated program. It's much easier to have to dodge one shortcut than many.

**Comment by emendelson** [comment]
Agreed - in fact, I was planning to try to make a second patch that would create a separate function that would enable the window-full-screen toggle key. But I don't think I have the skill to do this neatly.

**Comment by emendelson** [comment]
I can see that the logic of always allowing full-screen toggle would be something like this line in my patch:

`if (section->Get_bool("enabledosboxkeys")) {`

but with an OR condition - OR if the handler is named SwitchFullScreen.

I tried various ways of writing this line, but since I don't know what I'm doing, every try produced an error.

**Comment by joncampbell123** [comment]
The section Get_bool() function is probably throwing an exception because there is no such setting. Those are defined in src/dosbox.cpp which adds those settings, their types, and default values.

**Comment by emendelson** [comment]
Actually, the Get_bool() function works correctly, I think because the setting is defined in sdlmain.cpp, in the same way that all the other "[sdl]" section items, like"showmenu", are defined in sdlmain.cpp (around line 4290+), not in dosbox.cpp. What I can't figure out is to add an OR condition so that the SwitchFullScreen handler would work even if enabledoskeys=false.

**Comment by joncampbell123** [comment]
Try this:

Look at src/gui/sdl_mapper.cpp, at the definition of CHandlerEvent.

You can try adding the conditional blocking in the CHandlerEvent::Active() function to see if that helps.

**Comment by emendelson** [comment]
Yes, I see: I'll try that (probably not until I get back from some travel) and will report back. Thank you again.

**Comment by Shane32** [comment]
In #575 I suggested a new conf entry to select from multiple default mappings.  'none' could be one of them.
